components/sidebar_right/sidebar_right.tsx

```diff
@@ -129,26 +138,26 @@ export default class SidebarRight extends React.PureComponent {
         }
     }
 
-    onFinishTransition = (e) => {
+    onFinishTransition = (e: { propertyName: string }): void => {
         if (e.propertyName === 'transform') {
             this.setState({isOpened: this.props.isOpen});
         }
     }
 
-    onShrink = () => {
+    onShrink = (): void => {
         this.props.actions.setRhsExpanded(false);
     };
 
-    handleUpdateSearchTerms = (term) => {
+    handleUpdateSearchTerms = (term: string): void => {
         this.props.actions.updateSearchTerms(term);
         this.focusSearchBar();
     }
 
-    getSearchBarFocus = (focusSearchBar) => {
+    getSearchBarFocus = (focusSearchBar: unknown): void => {
```

Comment:
this seems weird, I think you have to do that because at some point it is being called with a function that return void  as an argument, would you be so kind to fix it so we can have a simpler signature?

Reply:
This is the simplest signature I can think of right now as TypeScript is growing too fast and it is really hard to keep up.

But, I don't understand how can I simplest this further because I have added a type for the argument and added a return type for the function. If I don't add return type then ESLint is showing errors for same.

Reply:
I didn't get what to define type for `focusSearchBar` that's why I defined the `unknown` type.

Reply:
I can also use `any` but over the time TypeScript introduced `unknown` as `any` skip the type checking which can turn into heavy errors.  So, I used `unknown` instead of `any`.

Reply:
sorry, maybe I should have been more explicit, but I don't really see the point of nesting a call that doesn't generate a result, wouldn't it make more sense to concatenate them?
```js
searchBarFocus();
search();
```
so the signature now would be `() => void`

Reply:
I found that `searchBarFocus` has been in `search_bar` as well as in `search` under the same signature of `getFocus`.